python/ray/data/grouped_dataset.py

```diff
@@ -341,17 +341,20 @@ def get_key_boundaries(block_accessor: BlockAccessor):
         # The batch is the entire block, because we have batch_size=None for
         # map_batches() below.
         def group_fn(batch):
-            block_accessor = BlockAccessor.for_block(batch)
+            block = BlockAccessor.batch_to_block(batch)
+            block_accessor = BlockAccessor.for_block(block)
             if self._key:
                 boundaries = get_key_boundaries(block_accessor)
             else:
                 boundaries = [block_accessor.num_rows()]
             builder = DelegatingBlockBuilder()
             start = 0
             for end in boundaries:
-                group = block_accessor.slice(start, end)
-                applied = fn(group)
-                builder.add_block(applied)
+                group_block = block_accessor.slice(start, end)
+                group_block_accessor = BlockAccessor.for_block(group_block)
+                group_batch = group_block_accessor.to_batch_format(batch_format)
```

Comment:
Briefly comment there is batch -> block -> batch trip needed here?

Reply:
@jianoaix - sounds good, added.